packages/api/src/controllers/dashboard.ts

```diff
@@ -23,6 +24,70 @@ function pickAlertsByTile(tiles: Tile[]) {
   }, {});
 }
 
+type TileForAlertSync = Pick<Tile, 'id'> & {
+  config?: Pick<Tile['config'], 'alert'> | { alert?: IAlert | AlertDocument };
+};
+
+function extractTileAlertData(tiles: TileForAlertSync[]): {
+  tileIds: Set<string>;
+  tileIdsWithAlerts: Set<string>;
+} {
+  const tileIds = new Set<string>();
+  const tileIdsWithAlerts = new Set<string>();
+
+  tiles.forEach(tile => {
+    tileIds.add(tile.id);
+    if (tile.config?.alert) {
+      tileIdsWithAlerts.add(tile.id);
+    }
+  });
+
+  return { tileIds, tileIdsWithAlerts };
```

Comment:
If you wanted to depend more on lodash (already a [dep](https://github.com/hyperdxio/hyperdx/blob/main/packages/app/package.json#L63)), then there are functions that could reduce the number of lines of code:
```
const [tileIds, tileIdsWithAlerts] = _.map(_.partition(tiles, 'config.alert'), 'id');
return { tileIds: new Set(tileIds), tileIdsWithAlerts: new Set(tileIdsWithAlerts) }
```
Yes, you need to know what map and partition do but they also handle all the null/unknown cases when doing things with path strings.

There's nothing wrong, just a conversation as we go forward.